fideslog/api/models/analytics_event.py

```diff
@@ -81,12 +81,15 @@ def check_not_an_email_address(cls, value: str) -> str:
         return value
 
     @validator("endpoint")
-    def check_no_hostname(cls, value: str) -> str:
+    def validate_endpoint_format(cls, value: str) -> str:
         """
-        Ensure that endpoints contain only the URL path.
+        Ensure that endpoint contains method and path, and that path component contains only the URL path.
         """
-
-        return urlparse(value).path
+        endpoint_components: List[str] = value.split(":",1)
+        assert (
+            len(endpoint_components) > 1
+        ), "endpoint must contain both http method and path, delimited by a colon"
+        return f"{endpoint_components[0]}: {urlparse(endpoint_components[1]).path}"
```

Comment:
This is a lot clearer, thanks! One thing, though, should we also support lowercase HTTP method, e.g `Get` or `get` in addition to `GET`?

Reply:
That's a good callout - no, I don't think we should. I think it would just create inconsistencies in the data, and it can be easily avoided if we just call `.upper()` on the method string as part of the validation.